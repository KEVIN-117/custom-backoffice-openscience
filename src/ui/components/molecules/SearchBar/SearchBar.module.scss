@import '../../../../globals/ads-variables.scss';

.search__bar {
  align-items: center;
  background-color: $--ads-colors-gray-10;
  border-radius: 4px;
  display: flex;
  padding: 10px;
  width: 100%;
}

.input__field {
  flex-grow: 1;
}

.magnifying__icon {
  cursor: pointer;
  margin-left: 8px;
}
