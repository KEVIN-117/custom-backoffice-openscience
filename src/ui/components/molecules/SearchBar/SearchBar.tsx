import React from 'react';
import styles from './SearchBar.module.scss';
import type { IProps } from './types/IProps';
import InputField from '../../atoms/InputField/InputField';
import MagnifyingGlass from '../../atoms/MagnifyingGlass/MagnifyingGlass';
import searchIcon from '../../../../assets/icons/search.svg?raw';

function SearchBar({ placeholder, onChange, onSearchClick }: IProps) {
  return (
    <div className={styles.search__bar}>
      <InputField className={styles.input__field} placeholder={placeholder} onChange={onChange} />
      <MagnifyingGlass
        icon={searchIcon}
        onClick={onSearchClick}
        className={styles.magnifying__icon}
      />
    </div>
  );
}

export default SearchBar;
