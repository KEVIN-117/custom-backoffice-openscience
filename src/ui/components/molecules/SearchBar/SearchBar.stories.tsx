import type { StoryObj, Meta } from '@storybook/react';
import SearchBar from './SearchBar';

const meta: Meta<typeof SearchBar> = {
  title: 'ui/components/molecules/SearchBar',
  component: SearchBar,
  argTypes: {
    onChange: { action: 'typed' },
    onSearchClick: { action: 'search clicked' },
  },
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    placeholder: 'Search...',
  },
};
