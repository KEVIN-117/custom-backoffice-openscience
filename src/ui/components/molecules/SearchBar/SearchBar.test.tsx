import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import SearchBar from './SearchBar';

describe('SearchBar', () => {
  it('calls onChange when typing in the input field', () => {
    let inputValue = '';
    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      inputValue = event.target.value;
    };

    const { getByPlaceholderText } = render(
      <SearchBar placeholder="Search" onChange={handleChange} />,
    );
    const input = getByPlaceholderText('Search');
    fireEvent.change(input, { target: { value: 'Test' } });

    expect(inputValue).toBe('Test');
  });

  it('calls onSearchClick when the search icon is clicked', () => {
    let clicked = false;
    const handleSearchClick = () => {
      clicked = true;
    };

    const { getByRole } = render(
      <SearchBar placeholder="Search" onChange={() => {}} onSearchClick={handleSearchClick} />,
    );
    const button = getByRole('button');
    fireEvent.click(button);

    expect(clicked).toBe(true);
  });
});
