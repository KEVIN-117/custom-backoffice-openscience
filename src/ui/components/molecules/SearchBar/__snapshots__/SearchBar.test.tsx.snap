// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar Snapshot > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="_searchBar_777775"
  >
    <input
      aria-label="Search input"
      class="_inputField_04b06d"
      data-testid="input-field"
      placeholder="Search..."
      type="text"
    />
    <button
      aria-label="Search Icon"
      class="_magnifyingButton_023b10"
    />
  </div>
</DocumentFragment>
`;
